web/modules/custom/joinup_migrate/joinup_migrate.post_update.php

```diff
@@ -159,3 +158,86 @@ function joinup_migrate_post_update_more_redirects() {
     }
   }
 }
+
+/**
+ * Fix distribution files with the same name [ISAICP-3968].
+ */
+function joinup_migrate_post_update_same_name_distribution_files() {
+  /** @var \Drupal\Core\File\FileSystemInterface $file_system */
+  $file_system = \Drupal::service('file_system');
+  /** @var \Drupal\Component\Uuid\UuidInterface $uuid_generator */
+  $uuid_generator = Drupal::service('uuid');
+  /** @var \Drupal\Core\Entity\ContentEntityStorageInterface $redirect_storage */
+  $redirect_storage = \Drupal::entityTypeManager()->getStorage('redirect');
+
+  $legacy_db = Database::getConnection('default', 'migrate');
+  $legacy_db_name = $legacy_db->getConnectionOptions()['database'];
+
+  /** @var \Drupal\Core\Database\Query\SelectInterface $query */
+  $query = Database::getConnection()->select('file_managed', 'f')
+    ->fields('f', ['fid', 'uri'])
+    ->fields('d6f', ['filename', 'filepath'])
+    ->orderBy('f.uri')
+    ->condition('f.uri', 'public://distribution/%', 'LIKE');
+  $query->join("$legacy_db_name.files", 'd6f', 'f.fid = d6f.fid');
+  $file_groups = [];
+  foreach ($query->execute()->fetchAll() as $row) {
```

Comment:
Maybe drop a comment to explain what is the result of the query and what is the structure we are building.

Reply:
Done.